packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh

```diff
@@ -157,41 +167,62 @@ done
 max_depth=10000
 max_iterations=10000
 smt_timeout=100000
-max_workers=7 # Set to 7 since the CI machine has 8 CPUs
-# workers is the minimum between max_workers and the length of test_list
-# unless no test arguments are provided, in which case we default to max_workers
+max_workers=16 # Set to 16 since there are 16 proofs to run
+# workers is the minimum
+# between max_workers and the length of test_list unless no test
+# arguments are provided, in which case we default to max_workers
 if [ "$CUSTOM_TESTS" == 0 ] && [ "$SCRIPT_TESTS" == false ]; then
   workers=${max_workers}
 else
   workers=$((${#test_list[@]}>max_workers ? max_workers : ${#test_list[@]}))
 fi
 reinit=--reinit
 reinit=
-break_on_calls=--no-break-on-calls
-# break_on_calls=
 break_every_step=--break-every-step
 break_every_step=
+break_on_calls=--break-on-calls
+break_on_calls=
 auto_abstract=--auto-abstract-gas
 auto_abstract=
 bug_report=--bug-report
 bug_report=
-use_booster=--use-booster
-# use_booster=
+use_booster=--no-use-booster
+use_booster=
 state_diff="./snapshots/state-diff/Kontrol-Deploy.json"
 
 #############
 # RUN TESTS #
 #############
+# Set up the trap to run the function on failure
+trap on_failure ERR INT TERM
+trap clean_docker EXIT
 conditionally_start_docker
 
+results=()
+# Run kontrol_build and store the result
 kontrol_build
+results[0]=$?
+
+# Run kontrol_prove and store the result
 kontrol_prove
+results[1]=$?
 
-dump_log_results
+get_log_results
 
-if [ "$LOCAL" == false ]; then
-    notif "Stopping docker container"
-    clean_docker
+# Now you can use ${results[0]} and ${results[1]}
+# to check the results of kontrol_build and kontrol_prove, respectively
+if [ ${results[0]} -ne 0 ] && [ ${results[1]} -ne 0 ]; then
+  echo "Kontrol Build and Prove Failed"
+  exit 1
+elif [ ${results[0]} -ne 0 ]; then
+  echo "Kontrol Build Failed"
+  exit 1
+elif [ ${results[1]} -ne 0 ]; then
+  echo "Kontrol Prove Failed"
+  exit 2
+  # Handle failure
+else
```

Comment:
Are these error strings the only thing logged, or do we also get e.g. error messages explaining the failures? I'm assuming these would still be present in the terminal as part of the regular kontrol logs, but the `Handle failure` comment sounds like it might be a TODO?

Reply:
Bad placement of the comment, should've been moved up above / removed. These catches are to make sure we capture failures and clean-up the env properly so we fetch all the appropriate files in the container before shutting down. No further planned TODO here as far as I'm aware.